core/services/ocr2/plugins/ocr2vrf/fixed_blockhash_provider.go

```diff
@@ -0,0 +1,109 @@
+package ocr2vrf
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rpc"
+	"github.com/smartcontractkit/ocr2vrf/types"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm/client"
+)
+
+// Implementation of Blockhashes interface that returns blockhashes with fixed-sized windows relative to current block height
+type FixedBlockhashProvider struct {
+	client client.Client
+	// end block = current head - waitBlocks
+	waitBlocks uint
+	// start block = current head - lookbackBlocks
+	lookbackBlocks uint
```

Comment:
Do we want this to be configurable? It seems like the implementation of this interface for EVM chains should do this:

```golang
// Return a list of the recent blocks whose hashes are currently cheaply
// verifiable onchain, in view of the local node. (On Ethereum, these are the
// last 256 blocks, verifiable via the BLOCKHASH opcode.)
```

which would suggest that `lookbackBlocks` would always be 256 and `waitBlocks` would always be zero right?

Reply:
great question. It seems like `waitBlocks` can always be 0 (return up to most recent blocks) since we check against ConfirmationDelay anyways [here](https://github.com/smartcontractkit/ocr2vrf/blob/main/internal/vrf/ocr.go#L65) 

`lookbackBlocks` will rarely need to go beyond 256 but I was thinking in case, we need to replay old requests, we can have this configurable especially for fast chains. 

where did you find the comments above? I couldn't find any RPC method that would return 256 blocks.

Reply:
Yeah the way you're doing it now is fine, AFAIK there is no singular method that can return the last 256 block headers.